ApiDoctor.Validation/OData/IODataNavigable.cs

```diff
@@ -112,7 +112,8 @@ public static IODataNavigable NavigateByFunction(this IODataNavigable source, st
             {
                 foreach (var m in matches)
                 {
-                    m.IsComposable = true;
+                    // if its the last segment, it may not really be composable. So default to what already is already there.
+                    m.IsComposable = !isLastSegment || m.IsComposable;
```

Comment:
```suggestion
                    m.IsComposable ||= !isLastSegment;
```

(or it might be |=)

Reply:
ah, disregard https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#compound-assignment

Reply:
Thanks, applied in the latest update.